components/messenger.rst

```diff
@@ -284,17 +285,24 @@ do is to write your own CSV receiver::
 
         public function get(): void
         {
-            $ordersFromCsv = $this->serializer->deserialize(file_get_contents($this->filePath), 'csv');
-
-            foreach ($ordersFromCsv as $orderFromCsv) {
-                $order = new NewOrder($orderFromCsv['id'], $orderFromCsv['account_id'], $orderFromCsv['amount']);
-
-                $envelope = new Envelope($order);
-
-                $handler($envelope);
+            // Receive the envelope according to your transport ($yourEnvelope here),
+            // in most cases, using a connection is the easiest solution.
+            
+            if (null === $yourEnvelope) {
+                return [];
             }
-
-            return [$envelope];
+            
+            try {
+                $envelope = $this->serializer->decode([
+                    'body' => $yourEnvelope['body'],
+                    'headers' => $yourEnvelope['headers'],
+                ]);
+            } catch (MessageDecodingFailedException $exception) {
+                $this->connection->reject($yourEnvelope['id']);
+                throw $exception;
+            }
+            
+            return [$yourEnvelope->with(new CustomStamp($yourEnvelope['id']);
```

Comment:
I think it should be:
`return [$envelope->with(new CustomStamp($yourEnvelope['id']);`